sklearn/tree/tree.py

```diff
@@ -510,6 +515,111 @@ def decision_path(self, X, check_input=True):
         X = self._validate_X_predict(X, check_input)
         return self.tree_.decision_path(X)
 
+    def prune_tree(self):
+        """Prunes tree using Minimal Cost-Complexity Pruning.
+
+        .. versionadded:: 0.21
+        """
+        check_is_fitted(self, 'tree_')
+
+        if self.alpha == 0.0:
+            return
+
+        n_nodes = self.tree_.node_count
+
+        # prior probability using weighted samples
+        samples = self.tree_.weighted_n_node_samples
+        prior = samples / samples[0]
+
+        # impurity weighted by prior
+        r_node = prior * self.tree_.impurity
+
+        # find parent node ids and leaves
+        child_l = self.tree_.children_left
+        child_r = self.tree_.children_right
+        parents = np.zeros(shape=n_nodes, dtype=np.intp)
+
+        # Uses uint8 to avoid casting when passing to generate_pruned_tree
+        leaves_in_subtree = np.zeros(shape=n_nodes, dtype=np.uint8)
+
+        stack = [(0, -1)]
+        while len(stack) > 0:
+            node_id, parent = stack.pop()
```

Comment:
`node_idx` to stay consistent with the rest of the function.
I would also suggest `parent_idx` instead of `parent`,

and the `parents` array could just be named `parent`.